index.bs

```diff
@@ -2690,23 +2690,15 @@ partial interface MLGraphBuilder {
 ### The slice() method ### {#api-mlgraphbuilder-slice}
 Produce a slice of the input tensor.
 <script type=idl>
-dictionary MLSliceOptions {
-  sequence<unsigned long> axes;
-};
-
 partial interface MLGraphBuilder {
-  MLOperand slice(MLOperand input, sequence<long> starts, sequence<long> sizes,
-                optional MLSliceOptions options = {});
+  MLOperand slice(MLOperand input, sequence<unsigned long> starts, sequence<unsigned long> sizes);
 };
 </script>
 <div algorithm=slice>
     **Arguments:**
         - *input*: an {{MLOperand}}. The input tensor.
-        - *starts*: a sequence of {{long}}. The starting indices to slice of the corresponding axes of the input shape. A negative index value is interpreted as counting back from the end. For example, the value -1
-        - *sizes*: a sequence of {{long}}. The lengths to slice of the corresponding axes of the input shape.
-            The length value of -1 selects all the remaining elements from the starting index of the given axis.
-        - *options*: an optional {{MLSliceOptions}}. The optional parameters of the operation.
-            - *axes*: a sequence of {{unsigned long}}. The dimensions of the input shape to which *starts* and *sizes* apply. The values in the sequence must be in the range [0, N-1] where N is the rank of input tensor. When not specified, the sequence is assumed to be [0, ..., N-1], e.g. [0,1,2] for a 3-D tensor.
+        - *starts*: a sequence of {{unsigned long}}. The sequence of unsigned integer values indicating the starting index to slice of each input dimension, of length N where N is the rank of the input tensor. For each dimension *d* of *input*, *starts[d]* indicates the starting index to slice in that dimension.
+        - *sizes*: a sequence of {{unsigned long}}. The sequence of unsigned integer values indicating the size to slice of each input dimension, of length N where N is the rank of the input tensor. For each dimension *d* of *input*, *sizes[d]* indicates the size to slice in that dimension. The size must not be 0 and must satisfy the constraint *starting index + size <= input size* in that dimension.
```

Comment:
>The size must not be 0

This may need to be allowed, given some models I've seen (depending on their inputs) might temporarily slice all their content to an empty tensor and then readd padding later. Additionally some object detection models can legitimately return 0 bounding rects as a valid result. So, WebNN in general shouldn't disallow them imo. The WebNN backends can always just skip those operators at a key choke point when seeing 0 sizes, since they are generally nops anyway (that's what ORT's DML EP does when it encounters empty tensors). Though, since supporting empty tensors affects more than just `slice`, that's probably better as a separate issue - no action expected right now. 👍

Reply:
@fdwr 

> given some models I've seen (depending on their inputs) might temporarily slice all their content to an empty tensor and then readd padding later. Additionally some object detection models can legitimately return 0 bounding rects as a valid result. So, WebNN in general shouldn't disallow them imo. 

Thanks for your inputs. Those sound like good use cases. It probably deserves a separate issue for WG discussion. 

WDYT?

Reply:
Yep, separate discussion.